pkg/pilot/genericpilot/controller.go

```diff
@@ -38,19 +40,23 @@ func (e *GenericPilot) processNextWorkItem() bool {
 	}
 	defer e.queue.Done(key)
 
-	if k, ok := key.(string); ok {
-		if err := e.sync(k); err != nil {
-			glog.Infof("Error syncing Pilot %v, requeuing: %v", key.(string), err)
-			e.queue.AddRateLimited(key)
-		} else {
-			e.queue.Forget(key)
-		}
+	if err := e.sync(key); err != nil {
+		glog.Infof("Error syncing Pilot %v, requeuing: %v", key.(string), err)
+		e.queue.AddRateLimited(key)
+	} else {
+		e.queue.Forget(key)
 	}
 
 	return true
 }
 
-func (g *GenericPilot) sync(key string) (err error) {
+func (g *GenericPilot) sync(obj interface{}) (err error) {
+	var key string
+	var ok bool
+	if key, ok = obj.(string); !ok {
+		glog.Errorf("Unexpected non-string item passed to sync: %#v", obj)
+		return nil
+	}
```

Comment:
❓ Seems like the signature should remain `sync(key string)` and perform this check in the caller....and just blow up, since this should always be `string` as far as I can see.

Thats how Kubernetes Deployment controller deals with it: https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/deployment/deployment_controller.go#L472